### ISSUE TITLE
Accuracy problem

### ISSUE THREAD

**Issue by MengXinChengXuYuan** [danbochman/Real-Time-Action-Recognition#3]
Hi
Thank you for your great work first!
However I got a question on the accuracy of the real time recognition
I just cloned your repo, modified some code to adapt to py 3.5 and deleted all code about training and evaling. 
After downloading your pretrained model, by running spatial_cnn_cpu.py, I can get the recognition result but the accuracy is low and the confidence score is only about 0.45 maximum, quite different from the demo video given in the repo.
Is that in the demo video you are running on a gpu, using both spatial cnn and motion cnn, and finally fused the result?
Or in spatial_cnn_cpu.py you just use rgb as input, but in spatial_cnn_gpu, optflow also?

**Comment by danbochman** [comment]
Hello!
Actually the scores you see on the demo video are computed using only the spatial_cnn_cpu script! this video was made to show it can be rather successful in real-time using only the spatial cnn. The optical flow windows you see beside the video are also computed in real time, but as you can see, not in very high quality, so adding them to the fusion will not increase accuracy.
For the fusion to increase accuracy, very high quality optical flow output should be created in real-time and I'm afraid it's very difficult to do (but it will work if you can do it). 
About your accuracy score... you're getting that from a live-feed? maybe you have a substantially different webcam, or different aspect ratio which warps the image when resizing, some cameras have an inherent "fish-eye" effect from the lens which can warp the image. There can be many reasons... but using the pre-trained weights with this repo should yield the exact same results you see in the demo video. I know this for certain as I've re-created these results on different laptops while presenting it on different occasions (even when using the laptop's camera). 
Let me know if I can be of further assistance!

**Comment by MengXinChengXuYuan** [comment]
> Hello!
> Actually the scores you see on the demo video are computed using only the spatial_cnn_cpu script! this video was made to show it can be rather successful in real-time using only the spatial cnn. The optical flow windows you see beside the video are also computed in real time, but as you can see, not in very high quality, so adding them to the fusion will not increase accuracy.
> For the fusion to increase accuracy, very high quality optical flow output should be created in real-time and I'm afraid it's very difficult to do (but it will work if you can do it).
> About your accuracy score... you're getting that from a live-feed? maybe you have a substantially different webcam, or different aspect ratio which warps the image when resizing, some cameras have an inherent "fish-eye" effect from the lens which can warp the image. There can be many reasons... but using the pre-trained weights with this repo should yield the exact same results you see in the demo video. I know this for certain as I've re-created these results on different laptops while presenting it on different occasions (even when using the laptop's camera).
> Let me know if I can be of further assistance!

Alright ... That's strange, I just used a common camera and running the script on windows/CPU, and the resolution seems to be the same as yours...

I will try to dig deeper to find the problem

**Comment by danbochman** [comment]
You mentioned you deleted the code for training and eval but specifically the evaluation scripts can help you debug your problem. You can download the UCF-101 dataset and see if you achieve 82% accuracy on the test set with the Spatial CNN. If you do get 82% percent, then most probably the difference is due to your inference setup.
If not, then your camera probably isn't the issue and something went wrong when you modified the code.
